### PR TITLE
feat: project updates table

### PR DESCRIPTION
**Pull request by beejjacobs**
## Describe your changes

Added DB schema, API endpoints, and basic table view for application project updates.

## Issue ticket number and link

https://pins-ds.atlassian.net/browse/BOAS-876

Maybe also be a ticket on the FO board.

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

**Comment by beejjacobs** [comment]
I've added fields to Project Updates that aren't used yet, but should allow for future features (e.g. setting a title to use internally, and welsh content).

File: apps/api/src/database/schema.prisma [hunk 1]
```diff
@@ -663,6 +665,37 @@ model ListedBuildingDetails {
   lpaQuestionnaire      LPAQuestionnaire? @relation(fields: [lpaQuestionnaireId], references: [id])
 }
 
+// ProjectUpdate represents some content in relation to a case (application/nsip) that may be of interest to
+// members of the public and those with an interest in that case. These updates are ultimate sent out to email
```
Comment (Philip-Philippou1) [inline apps/api/src/database/schema.prisma:669]: ultimately

File: apps/api/src/database/migrations/20230619130451_project_updates_sent/migration.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
+BEGIN TRY
```
Comment (Philip-Philippou1) [inline apps/api/src/database/migrations/20230619130451_project_updates_sent/migration.sql:1]: Is it worth just having 1 migration file for this or is the history useful?

File: apps/api/src/database/schema.prisma [hunk 3]
```diff
@@ -663,6 +665,37 @@ model ListedBuildingDetails {
...
+  // should this update be emailed to subscribers?
+  emailSubscribers  Boolean   @default(false)
+  // status of this update
+  status            String    @default("draft") // draft | published | unpublished | archived
+  // when was this project update changed to status "published"
+  datePublished     DateTime?
+  // when the (background) task to notify subscribers has been run for this update, this will be marked true
+  sentToSubscribers Boolean   @default(false)
+  // an internal title/name for this update
+  title             String?
```
Comment (Philip-Philippou1) [inline apps/api/src/database/schema.prisma:691]: Just wanted to confirm title is optional, I don't have full context so I'm guessing it is.

File: apps/api/src/server/applications/application/project-updates/project-updates.mapper.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @param {import('@prisma/client').ProjectUpdate} projectUpdate
+ * @returns {import('@pins/applications').ProjectUpdate}
+ */
+export function mapProjectUpdate(projectUpdate) {
+	return {
+		id: projectUpdate.id,
```
Comment (Philip-Philippou1) [inline apps/api/src/server/applications/application/project-updates/project-updates.mapper.js:7]: Should the get return all the fields such as title, sentToSubscribers, welsh...

File: apps/api/src/server/applications/application/project-updates/project-updates.controller.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * @type {import('express').RequestHandler}
+ */
+export async function getProjectUpdates(req, res) {
+	const page = Number(req.query.page) || DEFAULT_PAGE_NUMBER;
+	const pageSize = Number(req.query.pageSize) || DEFAULT_PAGE_SIZE;
+	const caseId = parseInt(req.params.id);
+	logger.debug({ caseId, page, pageSize }, 'getProjectUpdates');
+	let orderBy;
+
+	if (req.query.sortBy) {
```
Comment (Philip-Philippou1) [inline apps/api/src/server/applications/application/project-updates/project-updates.controller.js:17]: Looks like there are multiple implementations of sortBy, representations.controller.js also has one, my be good to re-use the sorting and paging logic in the future.

Comment (beejjacobs) [reply apps/api/src/server/applications/application/project-updates/project-updates.controller.js]: Consolidated into a util: https://github.com/Planning-Inspectorate/back-office/pull/992/commits/2fad469ef92a434ce459fe225020248d1593197e

File: apps/api/src/server/applications/application/project-updates/project-updates.mapper.js (hunk 4 above)
Comment (beejjacobs) [reply apps/api/src/server/applications/application/project-updates/project-updates.mapper.js:7]: Yeah, makes sense to let the consumer decide what they want: https://github.com/Planning-Inspectorate/back-office/pull/992/commits/ef8c4dcea4eafc27a7e0c57c77cf18df806970be

File: apps/api/src/database/schema.prisma (hunk 3 above)
Comment (beejjacobs) [reply apps/api/src/database/schema.prisma:692]: Yup, note added: https://github.com/Planning-Inspectorate/back-office/pull/992/commits/896ea08bb5ca4bfdc3dd1fc0970e30531b55c961

Internal title might become a field on the UI, but for MVP won't be there.

File: apps/api/src/database/migrations/20230619130451_project_updates_sent/migration.sql (hunk 2 above)
Comment (beejjacobs) [reply apps/api/src/database/migrations/20230619130451_project_updates_sent/migration.sql]: I don't think this kind of history is useful, combined: https://github.com/Planning-Inspectorate/back-office/pull/992/commits/a5efdba0d29a20661f82c69a64bb381a01dec384

File: apps/api/src/database/schema.prisma (hunk 1 above)
Comment (beejjacobs) [reply apps/api/src/database/schema.prisma]: Good spot: https://github.com/Planning-Inspectorate/back-office/pull/992/commits/896ea08bb5ca4bfdc3dd1fc0970e30531b55c961

**Review by Philip-Philippou1 — DISMISSED** [review]

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**